src/spectrum.rs

```diff
@@ -0,0 +1,397 @@
+use std::fmt;
+use std::ops::{Add, Sub, Mul, Div};
+
+use IntoColor;
+use flt;
+use num::{Float, FromPrimitive, Zero};
+use ordered_float::OrderedFloat;
+use white_point::WhitePoint;
+use xyz::Xyz;
+
+/// The smallest wavelength represented in the `Spectrum`
+/// data structure. This is the first value in the
+/// map's wavelength.
+pub const MIN_LAMBDA: usize = 360;
+
+/// The largest wavelength represented in the `Spectrum`
+/// data structure. This is the last value in the map's
+/// wavelength.
+#[allow(dead_code)]
+pub const MAX_LAMBDA: usize = 830;
+
+const SAMPLE_STEP: usize = 5;
+
+pub const N_SAMPLES: usize = (MAX_LAMBDA - MIN_LAMBDA) / SAMPLE_STEP + 1;
```

Comment:
It's something with the `N_` prefix... Perhaps just `SAMPLES` is good enough? Or is that ambiguous? `SECTRUM_LENGTH`? `SECTRUM_LEN`, to follow the `.len()` pattern?